From 9cb36770fef1bae1dca27dff6c45cc8a834d5872 Mon Sep 17 00:00:00 2001
From: Lucas Bickel <hairmare@purplehaze.ch>
Date: Fri, 4 May 2012 13:19:19 +0300
Subject: [PATCH 1/2] get eix cachefile location from eix

I want to keep the eix cache file in a single location for a vserver setup I'm doing. This change helps by allowing me to decide where I want to store the cache through default eix mechanisms by writing a config file with puppet.
---
 lib/puppet/provider/package/portage.rb |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/lib/puppet/provider/package/portage.rb b/lib/puppet/provider/package/portage.rb
index 591e9ea..4496db1 100644
--- a/lib/puppet/provider/package/portage.rb
+++ b/lib/puppet/provider/package/portage.rb
@@ -18,9 +18,10 @@ def self.instances
 
     version_format = "{last}<version>{}"
     search_format = "<category> <name> [<installedversions:LASTVERSION>] [<bestversion:LASTVERSION>] <homepage> <description>\n"
-
+    eix_cachefile = (eix "--print", "EIX_CACHEFILE").rstrip
+    
     begin
-      update_eix if !FileUtils.uptodate?("/var/cache/eix", %w{/usr/bin/eix /usr/portage/metadata/timestamp})
+      update_eix if !FileUtils.uptodate?(eix_cachefile, %w{/usr/bin/eix /usr/portage/metadata/timestamp})
 
       search_output = nil
       Puppet::Util.withenv :LASTVERSION => version_format do
@@ -76,12 +77,13 @@ def query
 
     version_format = "{last}<version>{}"
     search_format = "<category> <name> [<installedversions:LASTVERSION>] [<bestversion:LASTVERSION>] <homepage> <description>\n"
-
+    eix_cachefile = (eix "--print", "EIX_CACHEFILE").rstrip
+    
     search_field = package_name.count('/') > 0 ? "--category-name" : "--name"
     search_value = package_name
 
     begin
-      update_eix if !FileUtils.uptodate?("/var/cache/eix", %w{/usr/bin/eix /usr/portage/metadata/timestamp})
+      update_eix if !FileUtils.uptodate?(eix_cachefile, %w{/usr/bin/eix /usr/portage/metadata/timestamp})
 
       search_output = nil
       Puppet::Util.withenv :LASTVERSION => version_format do
-- 
1.7.10


From 7a281c4772a66f3295d9af821efed32068946bea Mon Sep 17 00:00:00 2001
From: Lucas Bickel <hairmare@purplehaze.ch>
Date: Fri, 4 May 2012 13:30:11 +0300
Subject: [PATCH 2/2] whitespace fix

---
 lib/puppet/provider/package/portage.rb |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/puppet/provider/package/portage.rb b/lib/puppet/provider/package/portage.rb
index 4496db1..9a2cd7b 100644
--- a/lib/puppet/provider/package/portage.rb
+++ b/lib/puppet/provider/package/portage.rb
@@ -19,7 +19,7 @@ def self.instances
     version_format = "{last}<version>{}"
     search_format = "<category> <name> [<installedversions:LASTVERSION>] [<bestversion:LASTVERSION>] <homepage> <description>\n"
     eix_cachefile = (eix "--print", "EIX_CACHEFILE").rstrip
-    
+
     begin
       update_eix if !FileUtils.uptodate?(eix_cachefile, %w{/usr/bin/eix /usr/portage/metadata/timestamp})
 
@@ -78,7 +78,7 @@ def query
     version_format = "{last}<version>{}"
     search_format = "<category> <name> [<installedversions:LASTVERSION>] [<bestversion:LASTVERSION>] <homepage> <description>\n"
     eix_cachefile = (eix "--print", "EIX_CACHEFILE").rstrip
-    
+
     search_field = package_name.count('/') > 0 ? "--category-name" : "--name"
     search_value = package_name
 
-- 
1.7.10

